src/query.ts

```diff
@@ -204,231 +243,174 @@ export class Query {
     }
+    return this;
   }
 
   /**
    * Sets the types to query in this query, call without
    * parameters to clear the setting
    *
-   * @param  {String | String[]}  type    The type or types to set
-   *                                      for this query
-   *
-   */
-  public setTypes(type?: string | string[]): void {
-    if (type) {
-      this.setParameter('type', type);
-    } else {
-      delete this.params.type;
-    }
-  }
-
-  /**
-   * Adds a type to the current types in this query
-   *
-   * @param  {String}   type    The type to be added for the search
+   * @param  type - The type or types to set for this query
    *
+   * @returns Query
    */
-  public addType(type: string): void {
-    let typeParam: string[] = [];
-
-    if (typeof this.params.type === 'string') {
-      typeParam.push(this.params.type);
-    } else if (this.params.type) {
-      typeParam = this.params.type;
-    }
-
-    typeParam.push(type);
-    this.setParameter('type', typeParam);
-  }
-
-  /**
-   * Remove a type from this query
-   *
-   * @param  {String}   type    The type to be removed for the search
-   *
-   */
-  public removeType(type: string): void {
-    if (this.params.type) {
-      if (isArray(this.params.type)) {
-        const index: number = this.params.type.indexOf(type);
-        if (index !== -1) {
-          (this.params.type as Array<unknown>).splice(1, index);
+  public addTypes(types?: string | string[]): Query {
+    if (types) {
+      this._dataTypes = new Set();
+      if (typeof types === 'string') {
+        this._dataTypes.add(types);
+      } else if (isArray(types)) {
+        for (const type of types) {
+          this._dataTypes.add(type);
         }
-      } else if (this.params.type === type) {
-        delete this.params.type;
+      } else {
+        throw new QueryValueError('Value error: Types value must be string or array of strings');
       }
+    } else {
+      delete this._dataTypes;
     }
+    return this;
   }
 
   /**
    * Sets the transformer, call it empty to reset it to null
    *
-   * @param  {String}   transformer   The transformer option to set
+   * @param  transformer - The transformer option to set
+   *
+   * @returns Query
    */
-  public transformer(transformer?: TransformerOptions): void {
+  public transformer(transformer?: string): Query {
     if (transformer) {
-      this.setParameter('transformer', transformer);
+      this._transformer = this._validateTransformer(transformer);
     } else {
-      delete this.params.transformer;
+      this._transformer = undefined;
     }
+    return this;
   }
 
   /**
    * Sets the timeout for the query, call it empty to reset
    *
-   * @param  {Number}   timeout       The timeoout for the call
+   * @param  timeout - The timeoout for the call
    *
    */
   public timeout(timeout?: number): void {
     if (timeout) {
-      this.setParameter('timeout', timeout);
+      this._timeout = timeout;
```

Comment:
Is there a default timeout in the service? Is it greater than `0`? If that's the case, this doesn't work when the passed value is `0`. For instance, you call `query.timeout(0)` and you get no timeout parameter.

Also, this uses `this._timeout` and `this.params.timeout` at the same time… 🤔